### PR TITLE
fix(WebAssemblyInterface): support vector/multi-component images

### PR DESCRIPTION
**Pull request by jadh4v**
Fix buffer size calculations and number of components for handling vector images.

### CODE REVIEW

File: include/itkOutputImage.h [hunk 1]
```diff
@@ -84,7 +84,7 @@ class ITK_TEMPLATE_EXPORT OutputImage
 
         const auto dataAddress = reinterpret_cast< size_t >( wasmImage->GetImage()->GetBufferPointer() );
         using ConvertPixelTraits = DefaultConvertPixelTraits<typename ImageType::PixelType>;
-        const auto dataSize = wasmImage->GetImage()->GetPixelContainer()->Size() * sizeof(typename ConvertPixelTraits::ComponentType) * ConvertPixelTraits::GetNumberOfComponents();
+        const auto dataSize = wasmImage->GetImage()->GetPixelContainer()->Size() * sizeof(typename ConvertPixelTraits::ComponentType);
```
Comment (jadh4v) [inline include/itkOutputImage.h:87]: `GetPixelContainer()->Size()` already includes number of components.

**Comment by jadh4v** [comment]
@thewtex I'm guessing we don't have any multi-component image already in `test/input`. So I might have to add a new multi-component image and either update `itkWasmImageInterfaceTest.cxx` or write a new test for this to test conversion in both directions (itkImage <--> WasmImage).

**Comment by thewtex** [comment]
> @thewtex I'm guessing we don't have any multi-component image already in `test/input`. So I might have to add a new multi-component image and either update `itkWasmImageInterfaceTest.cxx` or write a new test for this to test conversion in both directions (itkImage <--> WasmImage).

@jadh4v thanks for the patch!

Yes, could you please add a test, similar to:

  https://github.com/InsightSoftwareConsortium/ITK-Wasm/blob/main/test/itkWasmImageInterfaceTest.cxx

that uses *test/Input/apple.jpg* as an input and, `itk::VectorImage` as the image type?

**Comment by jadh4v** [comment]
My test still fails on image comparison. I think this maybe because of jpeg format??
```xml
Trying itk::WriteImage(convertedImage, outputImageFile)
<DartMeasurement name="ImageError apple.jpg" type="numeric/double">1934</DartMeasurement>
<DartMeasurement name="ImageError" type="numeric/double">1934</DartMeasurement>
<DartMeasurement name="ImageError Minimum" type="numeric/double">3</DartMeasurement>
<DartMeasurement name="ImageError Maximum" type="numeric/double">8</DartMeasurement>
<DartMeasurement name="ImageError Mean" type="numeric/double">3.42037</DartMeasurement>
<DartMeasurementFile name="DifferenceImage" type="image/png">/home/jadhav/code/native-itkwasm/ITK-Wasm-build/Testing/Temporary/WebAssemblyInterface/itkWasmVectorImageInterfaceTest.jpg.diff.png</DartMeasurementFile>
<DartMeasurementFile name="BaselineImage" type="image/png">/home/jadhav/code/native-itkwasm/ITK-Wasm-build/Testing/Temporary/WebAssemblyInterface/itkWasmVectorImageInterfaceTest.jpg.base.png</DartMeasurementFile>
<DartMeasurementFile name="TestImage" type="image/png">/home/jadhav/code/native-itkwasm/ITK-Wasm-build/Testing/Temporary/WebAssemblyInterface/itkWasmVectorImageInterfaceTest.jpg.test.png</DartMeasurementFile>
<DartMeasurement name="BaselineImageName" type="text/string">apple.jpg</DartMeasurement>
```

**Comment by thewtex** [comment]
@jadh4v thanks!

> My test still fails on image comparison. I think this maybe because of jpeg format??

Yes, we can use .mha -- I will create a patch, add it to the beach. I will build / push the docker images and update the branch according.

**Review by thewtex — APPROVED** [review]
:clap:

**Pull request merged**